### PR TITLE
Refactor configuration

### PR DESCRIPTION
**Pull request by nikolaystanishev**

### CODE REVIEW

**Comment by nikolaystanishev** [comment]
[SP21: Refactor all gpf configuration](https://trello.com/c/kNeaXrAn/976-sp21-refactor-all-gpf-configuration)

File: dae/dae/studies/study.py [hunk 1]
```diff
@@ -9,38 +9,70 @@ def __init__(self, config):
...
 
     @property
-    def families(self):
-        raise NotImplementedError()
+    def pub_meds(self):
+        return [self.config.pub_med] if self.config.pub_med else []
+
+    @property
+    def study_types(self):
+        print(self.config.keys())
```
Comment (lchorbadjiev) [inline dae/dae/studies/study.py]: remove debug print

File: dae/dae/studies/study_sections_config.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from dae.studies.people_group_config_parser import PeopleGroupConfigParser
+from dae.studies.genotype_browser_config_parser import \
+    GenotypeBrowserConfigParser
+
+
+class StudySectionsConfigMixin(object):
```
Comment (lchorbadjiev) [inline dae/dae/studies/study_sections_config.py]: probably should be moved into StudyConfigParserBase

File: dae/dae/studies/tests/test_dataset_config.py [hunk 3]
```diff
@@ -14,12 +12,12 @@ def test_dataset_definitions_simple(dataset_definitions):
...
-    ("phenotypeGenotypeTool", False),
+    ("phenotypeTool", False),
     ("phenotypeBrowser", False),
     ("year", ""),
-    ("years", []),
-    ("pub_med", ""),
-    ("pub_meds", [])
+    # ("years", []),
+    # ("pub_med", ""),
+    # ("pub_meds", [])
```
Comment (lchorbadjiev) [inline dae/dae/studies/tests/test_dataset_config.py]: check if these work

File: dae/dae/configuration/dae_config_parser.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
+import os
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/configuration/dae_config_parser.py]: It is renamed to config_parser_base and configuration file is renamed to dae_config_parser

File: dae/dae/studies/study.py (hunk 1 above)
Comment (nikolaystanishev) [reply dae/dae/studies/study.py]: it is removed

File: dae/dae/configuration/dae_config_parser.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+        )
+
+        config['config_file'] = config_file
+
+        return config
+
+    @classmethod
+    def _collect_config_paths(cls, dirname, fail_silently=False):
+        config_paths = []
+        print(dirname)
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/configuration/dae_config_parser.py]: it is removed

File: dae/dae/VariantAnnotation.py [hunk 6]
```diff
@@ -1,86 +0,0 @@
-#!/bin/env python
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/VariantAnnotation.py]: Now it is moved to tools and it is renamed to variant_effect_annotator.py

File: dae/dae/studies/study_sections_config.py (hunk 2 above)
Comment (nikolaystanishev) [reply dae/dae/studies/study_sections_config.py]: I moved all of his methods to study config.

File: dae/dae/annotation/tools/score_annotator.py [hunk 7]
```diff
@@ -29,7 +29,7 @@ def __init__(self, config):
 
     def _init_score_file(self):
         assert self.config.options.scores_file, \
-            [self.config.name, self.config.annotator_name]
+            [self.config.annotator, self.config.annotator]
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/annotation/tools/score_annotator.py]: debug message now is only `self.config.annotator`

File: dae/dae/annotation/tools/utils.py [hunk 8]
```diff
@@ -1,3 +1,35 @@
+from importlib import import_module
+
+
+class AnnotatorInitializer(object):
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/annotation/tools/utils.py]: i renamed it to AnnotatorFactory and instantiate method become make_annotator

File: dae/dae/annotation/tools/annotator_base.py [hunk 9]
```diff
@@ -213,14 +213,14 @@ class VariantAnnotatorBase(AnnotatorBase):
...
 
-        assert isinstance(config, VariantAnnotatorConfig)
+        assert isinstance(config, Box)
 
         self.genome = None
 
         if self.config.options.vcf:
             self.variant_builder = VCFBuilder(self.config, self.genome)
         else:
-            self.genome = GenomeAccess.openRef(self.config.genome_file)
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/annotation/tools/annotator_base.py]: I made trello card and add @iordanivanov in it.

File: dae/dae/annotation/tools/effect_annotator.py [hunk 10]
```diff
@@ -43,7 +43,7 @@ def _init_effect_annotator(
                     assert self.config.options.Graw is not None
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/annotation/tools/effect_annotator.py]: I made trello card and add @iordanivanov in it.

File: dae/dae/annotation/tools/annotator_config.py [hunk 11]
```diff
@@ -1,141 +1,166 @@
...
+            if 'values' not in defaults:
+                defaults['values'] = {}
+            defaults['values'][f'options.{key}'] = option
+
+        config = super(AnnotationConfigParser, cls).read_file_configuration(
+            config_file, work_dir, defaults
+        )
 
-class AnnotatorConfig(object):
+        config.options = options
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/annotation/tools/annotator_config.py]: with this I set pipeline annotator options

File: dae/dae/pheno_browser/prepare_data.py [hunk 12]
```diff
@@ -242,22 +242,31 @@ def handle_measure(self, measure):
...
             if not reg_measure:
                 continue
-            if self.pheno_regressions.has_measure(measure.measure_name,
-                                                  measure.instrument_name):
+            if self._has_regression_measure(
+                    measure.measure_name, measure.instrument_name):
                 continue
 
-            res['regression_id'] = reg_id
+            res['regression_id'] = reg.id
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/pheno_browser/prepare_data.py]: It had a problem. Regression doesn't have id. With fixing that I fixed one other small bug.

File: dae/dae/studies/tests/test_dataset_config.py (hunk 3 above)
Comment (nikolaystanishev) [reply dae/dae/studies/tests/test_dataset_config.py]: I fix them and now they are working

File: dae/dae/studies/study_wrapper.py [hunk 13]
```diff
@@ -336,32 +323,31 @@ def _get_pheno_filter_constraints(self, pheno_filter):
         return set(selection['selection'])
 
     def _add_people_with_phenotype(self, kwargs):
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/studies/study_wrapper.py]: it is renamed

File: dae/dae/studies/tests/fixtures/datasets/quads_composite.conf [hunk 14]
```diff
@@ -63,7 +63,7 @@ genotype.geneeffect.source = geneEffect
 genotype.effectdetails.name = effect details
 genotype.effectdetails.source = effectDetails
 
-genotype.columns=family,phenotype,variant,best,
+selectedGenotypeValues=family,phenotype,variant,best,
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/studies/tests/fixtures/datasets/quads_composite.conf]: it is renamed to selectedGenotypeColumnValues

File: dae/dae/studies/tests/test_dataset_facade.py [hunk 15]
```diff
@@ -1,5 +1,5 @@
-def test_dataset_facade_simple(dataset_facade):
```
[the remark this answers is not in the document]
Comment (nikolaystanishev) [reply dae/dae/studies/tests/test_dataset_facade.py]: tests from this file are moved to variants db test

**Pull request merged**